### PR TITLE
Try to deprecate unstable python memory optimize

### PR DESCRIPTION
**Pull request by sneaxiy**
We have enabled [inplace](https://github.com/PaddlePaddle/Paddle/pull/17911) and [GC](https://github.com/PaddlePaddle/Paddle/pull/18836) by default. We do not need the legacy `fluid.memory_optimize` interface,  since this API is not stable.

After this PR, `fluid.memory_optimize` does not take any effect on the program.

CE Debug has passed. See [here](http://ce.paddlepaddle.org:8080/viewLog.html?buildId=54897&buildTypeId=PaddleModesl_ModelDebug_Model11Build&tab=buildLog). `dcgan_d_train_cost` fails because of the previous PR not this one.

### CODE REVIEW

**Review by chengduoZH — DISMISSED** [review]
LGTM

File: python/paddle/fluid/tests/unittests/parallel_executor_test_base.py [hunk 1]
```diff
@@ -68,14 +67,11 @@ def run_executor(exe, binary, feed, fetch_list):
             loss = method(use_feed=feed_dict is not None)
             # NOTE(zjl): memory_optimize/inplace pass would not require 
             # that loss.persistable = True 
-            loss.persistable = memory_opt
+            loss.persistable = False
```
Comment (chengduoZH) [inline python/paddle/fluid/tests/unittests/parallel_executor_test_base.py]: Why do you set `False` here?

**Review by chengduoZH — DISMISSED** [review]
LGTM

File: python/paddle/fluid/tests/unittests/parallel_executor_test_base.py (hunk 1 above)
Comment (sneaxiy) [reply python/paddle/fluid/tests/unittests/parallel_executor_test_base.py]: Because all memory optimization strategies do not need `loss.persistable = True`. I set `loss.persistable = False` intentionally to test whether the memory optimization strategy is right.

**Review by luotao1 — APPROVED** [review]
LGTM

File: python/paddle/fluid/tests/unittests/test_deprecated_memory_optimize_interfaces.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
```
Comment (luotao1) [inline python/paddle/fluid/tests/unittests/test_deprecated_memory_optimize_interfaces.py:1]: 请问什么时候会完全删除？包括删除`test_deprecated_memory_optimize_interfaces.py`?

Comment (sneaxiy) [reply python/paddle/fluid/tests/unittests/test_deprecated_memory_optimize_interfaces.py:1]: 接口壳子应该得保留一段时间，预计1.6会作为缓冲期，1.6以后接口可直接删除。
